lib/client/client.go

```diff
@@ -186,8 +186,22 @@ func (proxy *ProxyClient) ReissueUserCerts(ctx context.Context, params ReissuePa
 		return trace.Wrap(err)
 	}
 
+	localKeyAgent := proxy.localAgent()
+
+	// The re-issued certificate may contain new roles that change our
+	// permissions with *any* service (e.g. Kubernetes, Database) in the
+	// cluster. We want to make sure no cached certificates are preserved
+	// once we have the re-issued certificate, forcing each service to get
+	// a new certificate (with the potentially updated roles) on first
+	// access, so we explicitly delete them all.
+	log.Debugf("Deleting all user certs...")
+	if err = localKeyAgent.DeleteUserCerts(params.RouteToCluster, WithAllCerts...); err != nil {
+		log.Errorf("Deleting user certs failed")
```

Comment:
Assuming writing these logs is really needed here:
```suggestion
		log.WithError(err).Errorf("Deleting user certs failed.")
```

Reply:
Refactored out of existence